### PR TITLE
Add function call params to invocation params

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: langchain/chat_models/azure_openai.py [hunk 1]
```diff
@@ -121,12 +121,22 @@ def _identifying_params(self) -> Mapping[str, Any]:
...
-    def _invocation_params(self) -> Mapping[str, Any]:
+    def _client_params(self) -> Dict[str, Any]:
+        """Get the config params used for the openai client."""
         openai_creds = {
             "api_type": self.openai_api_type,
             "api_version": self.openai_api_version,
         }
-        return {**openai_creds, **super()._invocation_params}
+        return {**super()._client_params, **openai_creds}
+
```
Comment (hinthornw) [inline langchain/chat_models/azure_openai.py:131]: Weired that we had `_invocation_params` and `_get_invocation_params` that each served different purposes:
1. `_get_invocation_params` was used in callbacks and for cache checking
2. `_invocation_params` used for actually passing params to the openai and azure openai client

Some other options would be:
- Always do `_get_invocation_params` but then either don't call `super()` or change the `self.dict()` call to filter out the `_type` arguments 
- Always do `_get_invocation_params` but then add the api key in a separate method

File: langchain/chat_models/openai.py [hunk 2]
```diff
@@ -453,6 +454,22 @@ def _invocation_params(self) -> Mapping[str, Any]:
             openai.proxy = {"http": self.openai_proxy, "https": self.openai_proxy}  # type: ignore[assignment]  # noqa: E501
         return {**openai_creds, **self._default_params}
 
+    def _get_invocation_params(
+        self, stop: Optional[List[str]] = None, **kwargs: Any
+    ) -> Dict[str, Any]:
+        """Get the parameters used to invoke the model FOR THE CALLBACKS."""
+        functions = kwargs.get("functions") or (
+            [format_tool_to_openai_function(tool) for tool in kwargs["tools"]]
```
Comment (agola11) [inline langchain/chat_models/openai.py:462]: confused by this logic. Why would tools get passed here?

Comment (vowelparrot) [reply langchain/chat_models/openai.py:462]: Yeah agreed. Was a blind copy

Comment (hinthornw) [reply langchain/chat_models/openai.py:462]: Good catch. Was cargo culting the JS logic

**Pull request merged**